version/version.go

```diff
@@ -0,0 +1,43 @@
+package version
```

Comment:
It's better if we perform this task of setting ldflag variables from main, and then set them in the config and pass them to the server struct. And then we can return them in the ping response.

Creating a separate package just to have a single function and some variables is probably an overkill.

Reply:
That's the exact reason I did put them in their own package. Adding version metadata to Config to pass it around does not make much sense to me. These information are not _configuration_, rather metadata of the binary.

Also I don't quite like adding random top level variables in main just for the sake of ldflags to work.

I'm 4/5 to keep it like this but open to change it.

Reply:
Ok.

Reply:
Since we already have other assets we can include another one like `version.json` and pass all these ldflags into it at `make build` stage. Then we can just read it whenever we hit the ping handler. In fact, we can simply read it binary and attach to response as `json.RawMessage`. With this we won't need to manage `version.Info` struct at all.

Reply:
Since version information is more of metadata than asset I think it'd be better to keep it in `version.go` file and let the ldflags  - on build time - inject the values to it.